hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ServerCrashProcedure.java

```diff
@@ -482,8 +482,16 @@ private void assignRegions(MasterProcedureEnv env, List<RegionInfo> regions) thr
           regionNode.getProcedure().serverCrashed(env, regionNode, getServerName());
           continue;
         }
-        if (env.getMasterServices().getTableStateManager().isTableState(regionNode.getTable(),
-          TableState.State.DISABLING, TableState.State.DISABLED)) {
+        if (env.getMasterServices().getTableStateManager()
+          .isTableState(regionNode.getTable(), TableState.State.DISABLING)) {
+          env.getAssignmentManager().regionClosedAbnormally(regionNode);
+          LOG.info("{} found table disabling for region {}, set it state to ABNORMALLY_CLOSED.",
+            this, regionNode);
+          continue;
+        }
+        if (env.getMasterServices().getTableStateManager()
```

Comment:
And also here, described why we do not need to set the state. I think the state here should already be CLOSED? And I wonder whether we could reach here, since the region is sitll in OPEN or OPENING state but the server recorded in AM is  already dead so we can not close it in TRSP...

Reply:
ok